src/Common/tests/System/Net/RemoteServerQuery.cs

```diff
@@ -0,0 +1,75 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Net.Test.Common
+{
+    internal class RemoteServerException : Exception
+    {
+        private const string Description = "Likely external issue with remote server";
+
+        public RemoteServerException()
+            : this(null, null) { }
+
+        public RemoteServerException(string server)
+            : this(server, null) { }
+
+        public RemoteServerException(string server, Exception inner)
+            : base(GetMessage(server), inner) { }
+
+        public static string GetMessage(string server)
```

Comment:
Looks like helper method, do we need it public?

Reply:
This is not exposed in any api or anything, the whole class in internal. But yeah it can be private.